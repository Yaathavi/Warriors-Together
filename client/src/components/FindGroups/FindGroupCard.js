import React from "react";
import history from "../Navigation/history";

export let groupID = -1;
export let groupContent = {};

function FindGroupCard(props) {
  const onClick = () => {
    //If the user clicks on the group, go to the group page
    console.log("open", props.title, " with id", groupID);
    groupID = props.groupID;
    groupContent = {
      title: props.title,
      members: props.members,
      description: props.description,
      categories: props.categories,
    };
    history.push("/group");
  };

  return (
    <div
      className="flex flex-col w-1/4 mr-10 my-4 p-8 cursor-pointer"
      style={{
        boxShadow: "rgba(236, 236, 236, 0.44) 0px 0px 15px",
        border: "1px solid #f8f4ef",
        background: "#fff",
      }}
      onClick={() => onClick()}
    >
      <h2 className="text-xl font-semibold">{props.title}</h2>
      <p className="mt-1">
        {props.members} members | {props.categories}
      </p>
      <p className="mt-4">{props.description}</p>
    </div>
  );
}

export default FindGroupCard;
