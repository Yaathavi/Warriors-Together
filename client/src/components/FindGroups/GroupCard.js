import React from "react";

function GroupCard() {
  return (
    <div className="flex flex-col w-1/4 bg-gray-300 mr-10 my-4 p-8">
      <h2 className="text-xl">Group Name</h2>
      <p className="mt-1">12 members | sports, social</p>
      <p className="mt-4">
        Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod
        tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim
        veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea
        commodo consequat.
      </p>
    </div>
  );
}

export default GroupCard;
