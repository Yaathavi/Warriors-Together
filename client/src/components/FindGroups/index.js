import React, { useEffect } from "react";
import Header3 from "../Header/header3";
import HeaderDefault from "../Header/HeaderDefault";
import GroupCard from "./GroupCard";
import history from "../Navigation/history";

function FindGroups() {
  useEffect(() => {
    let authToken = sessionStorage.getItem("Auth Token");

    if (!authToken) {
      history.push("/login");
    }
  }, []);

  return (
    <div className="flex flex-col min-h-screen overflow-hidden bg-gradient-to-b from-gray-50 via-gray-100 to-gray-200">
      <HeaderDefault thisPage="findgroups" />
      <div className="mr-20 ml-20 flex flex-col">
        <h1 className="text-4xl font-semibold">Find Groups</h1>
        <div className="flex flex-row">
          <div class="m-5 mx-0 inline-block relative w-64">
            <select class="block appearance-none w-full bg-white border border-gray-400 hover:border-gray-500 px-4 py-2 pr-8 rounded shadow leading-tight focus:outline-none focus:shadow-outline">
              <option selected disabled>
                Filter by Interests
              </option>
              <option>Option 1</option>
              <option>Option 2</option>
            </select>
            <div class="pointer-events-none absolute inset-y-0 right-0 flex items-center px-2 text-gray-700">
              <svg
                class="fill-current h-4 w-4"
                xmlns="http://www.w3.org/2000/svg"
                viewBox="0 0 20 20"
              >
                <path d="M9.293 12.95l.707.707L15.657 8l-1.414-1.414L10 10.828 5.757 6.586 4.343 8z" />
              </svg>
            </div>
          </div>
          <div class="mt-5 inline-block relative w-64 mx-5">
            <select class="block appearance-none w-full bg-white border border-gray-400 hover:border-gray-500 px-4 py-2 pr-8 rounded shadow leading-tight focus:outline-none focus:shadow-outline">
              <option selected disabled>
                Sort By
              </option>
              <option>Option 1</option>
              <option>Option 2</option>
            </select>
            <div class="pointer-events-none absolute inset-y-0 right-0 flex items-center px-2 text-gray-700">
              <svg
                class="fill-current h-4 w-4"
                xmlns="http://www.w3.org/2000/svg"
                viewBox="0 0 20 20"
              >
                <path d="M9.293 12.95l.707.707L15.657 8l-1.414-1.414L10 10.828 5.757 6.586 4.343 8z" />
              </svg>
            </div>
          </div>
        </div>
        <div className="flex flex-wrap mt-5 justify-start">
          <GroupCard />
          <GroupCard />
          <GroupCard />
          <GroupCard />
          <GroupCard />
          <GroupCard />
        </div>
      </div>
    </div>
  );
}

export default FindGroups;
