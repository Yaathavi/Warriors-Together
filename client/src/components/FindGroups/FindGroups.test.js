/**
 * @jest-environment jsdom
 */

import React from "react";
import renderer from "react-test-renderer";
import FindGroups from "./index";

it("Renders the correct component", () => {
  const component = renderer.create(
    <FindGroups />
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
