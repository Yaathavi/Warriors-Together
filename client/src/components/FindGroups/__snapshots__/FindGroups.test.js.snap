// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders the correct component 1`] = `
<div
  className="flex flex-col min-h-screen overflow-hidden bg-gradient-to-b from-gray-50 via-gray-100 to-gray-200 mb-10"
  style={
    {
      "background": "#fbfbfa",
    }
  }
>
  <div>
    <nav
      class="w-full z-30 top-0 text-blue-1000 py-1 lg:py-6"
    >
      <div
        class="w-full flex flex-wrap justify-between"
      >
        <span>
          <span
            class="hover:no-underline font-bold text-2xl lg:text-4xl flex flex-row items-center"
          >
            <a
              class="hover:no-underline font-bold text-2xl lg:text-4xl flex flex-row items-center"
              href="/"
            >
              <img
                className="w-20 ml-20 mr-5"
                src="./logo.png"
              />
              <p
                className="text-4xl text-blue-1000 font-bold"
              >
                 
                WarriorsTogether
                 
              </p>
            </a>
          </span>
        </span>
        <span
          className="mr-20 text-center my-auto"
        >
          <a
            class="inline-block py-2 px-4"
            href="/mygroups"
          >
            <p
              className="text-lg font-semibold"
            >
              My Groups
            </p>
          </a>
          <p
            class="inline-block py-2 px-4 underline decoration-2"
          >
            <p
              className="text-lg font-semibold"
            >
              Find Groups
            </p>
          </p>
          <a
            class="inline-block py-2 px-4"
            href="/settings"
          >
            <p
              className="text-lg font-semibold"
            >
              Settings
            </p>
          </a>
        </span>
      </div>
    </nav>
  </div>
  <div
    className="mr-20 ml-20 flex flex-col"
  >
    <h1
      className="text-4xl font-semibold"
    >
      Find Groups
    </h1>
    <div
      className="flex flex-row"
    >
      <div
        class="m-5 mx-0 inline-block relative w-64"
      >
        <select
          class="block appearance-none w-full bg-white border border-gray-400 hover:border-gray-500 px-4 py-2 pr-8 rounded shadow leading-tight focus:outline-none focus:shadow-outline"
          onChange={[Function]}
        >
          <option
            disabled={true}
            selected={true}
          >
            Filter by Interests
          </option>
          <option>
            All
          </option>
          <option>
            Engineering
          </option>
          <option>
            Math
          </option>
          <option>
            Science
          </option>
          <option>
            Arts
          </option>
          <option>
            Tech
          </option>
          <option>
            Food
          </option>
          <option>
            Gaming
          </option>
          <option>
            Clothes
          </option>
          <option>
            Fashion
          </option>
          <option>
            Music
          </option>
          <option>
            Movies
          </option>
          <option>
            Makeup
          </option>
          <option>
            Social
          </option>
          <option>
            Accounting
          </option>
          <option>
            Languages
          </option>
          <option>
            Dancing
          </option>
          <option>
            Singing
          </option>
          <option>
            Athletics
          </option>
        </select>
        <div
          class="pointer-events-none absolute inset-y-0 right-0 flex items-center px-2 text-gray-700"
        >
          <svg
            class="fill-current h-4 w-4"
            viewBox="0 0 20 20"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M9.293 12.95l.707.707L15.657 8l-1.414-1.414L10 10.828 5.757 6.586 4.343 8z"
            />
          </svg>
        </div>
      </div>
      <div
        class="mt-5 inline-block relative w-64 mx-5"
      >
        <select
          class="block appearance-none w-full bg-white border border-gray-400 hover:border-gray-500 px-4 py-2 pr-8 rounded shadow leading-tight focus:outline-none focus:shadow-outline"
          onChange={[Function]}
        >
          <option
            disabled={true}
            selected={true}
          >
            Sort By
          </option>
          <option>
            Name
          </option>
          <option>
            Members
          </option>
          <option>
            Joined
          </option>
        </select>
        <div
          class="pointer-events-none absolute inset-y-0 right-0 flex items-center px-2 text-gray-700"
        >
          <svg
            class="fill-current h-4 w-4"
            viewBox="0 0 20 20"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M9.293 12.95l.707.707L15.657 8l-1.414-1.414L10 10.828 5.757 6.586 4.343 8z"
            />
          </svg>
        </div>
      </div>
    </div>
    <div
      className="flex flex-wrap mt-5 justify-start"
    />
  </div>
</div>
`;
